async-nats/src/jetstream/consumer/pull.rs

```diff
@@ -785,13 +808,23 @@ impl<'a> futures::Stream for Ordered<'a> {
             self.stream = None;
             self.create_stream = Some(Box::pin({
                 let context = self.context.clone();
-                let config = self.consumer.clone().into();
+                let config = self.consumer.clone();
                 let stream_name = self.stream_name.clone();
                 let consumer_name = self.consumer_name.clone();
-                let sequence = self.consumer_sequence;
+                let sequence = self.stream_sequence;
                 async move {
-                    recreate_consumer_stream(context, config, stream_name, consumer_name, sequence)
-                        .await
+                    let strategy =
+                        tokio_retry::strategy::ExponentialBackoff::from_millis(500).take(5);
+                    tokio_retry::Retry::spawn(strategy, || {
+                        recreate_consumer_stream(
```

Comment:
I wonder whether we should retry the whole `delete_consumer -> create_consumer -> create_iterator` sequence. I think deleting consumer might be a one off (or even retry on timeout), but then we should retry creating consumer separately. Right now you're retrying delete consumer as well even if only creating consumer failed.

Reply:
Also, I wonder - according to the documentation of `from_millis:
```
Constructs a new exponential back-off strategy, given a base duration in milliseconds.

The resulting duration is calculated by taking the base to the n-th power, where n denotes the number of past attempts.
```

Doesn't that mean that the second retry will be done after 500 * 500ms? So after 250s? Shouldn't that be e.g.:
```rust
tokio_retry::strategy::ExponentialBackoff::from_millis(2).factor(250).take(5);
```
That would give us a backoff with `base == 2` and initial delay 500ms:

```rust
#[test]
fn base_2_initial_500ms() {
    let mut s = ExponentialBackoff::from_millis(2).factor(250);

    assert_eq!(s.next(), Some(Duration::from_millis(500)));
    assert_eq!(s.next(), Some(Duration::from_millis(1000)));
    assert_eq!(s.next(), Some(Duration::from_millis(2000)));
}
```

Reply:
I also don't think it's good to hard code 5 attempts - user should be able to set this to infinite.

Reply:
I will check the backoff fator and timeout when it happens 5 times.

However I disagree on infinite - My idea is that ordered consumer should be durable, but it should also error if it cannot recover in some sensible time. I don't think that infinite in this case is a good idea, as this can be a footgun IMO.

Reply:
Even if you couldn't make it infinite - backoff with base 2 and 500ms delay gives. a maximum retry attempt after 500 + 1000 + 2000 + 4000 + 8000ms, so after 15 seconds - I imagine restarting a js server with huge streams, network latencies etc might take longer.

Reply:
That is true, however I really do not like hanging consumer for  so long, without giving any error to the user and retry in the background.